lib/puppet/http/client.rb

```diff
@@ -144,6 +249,9 @@ def delete(url, headers: {}, params: {}, options: {})
     end
   end
 
+  #
+  # Close persistent connections in the pool
+  #
   def close
     @pool.close
   end
```

Comment:
Everything below here is `private`, so I skipped documenting it. LMK if you want me to go back and add documentation